modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -338,7 +345,12 @@ public PartitionReplicaListener(
 
         cursors = new ConcurrentSkipListMap<>(IgniteUuid.globalOrderComparator());
 
-        schemaCompatValidator = new SchemaCompatibilityValidator(validationSchemasSource, catalogService, schemaSyncService);
+        schemaCompatValidator = new SchemaCompatibilityValidator(
```

Comment:
As you can see from the comments above I don't like an idea of mixing business logic inside PartitionReplicaListener with tread switching one. I'd say that we should propagate decorated schemaSyncService, raft client that will switch tread on replica enter.